core/src/test/scala/zio/logging/LoggerSpec.scala

```diff
@@ -4,159 +4,151 @@ import java.time.OffsetDateTime
 
 import zio.test.Assertion._
 import zio.test._
-import zio.test.environment.TestClock
 import zio.{ Ref, UIO, ZIO }
 
-object LoggerSpec
-    extends DefaultRunnableSpec({
+object LoggerSpec extends DefaultRunnableSpec {
 
-      case class TestLogger(ref: Ref[Vector[(LogContext, String)]], logger: Logger) extends Logger {
-        override def locally[R, E, A1](f: LogContext => LogContext)(zio: ZIO[R, E, A1]): ZIO[R, E, A1] =
-          logger.locally(f)(zio)
-        override def log(line: => String): UIO[Unit] =
-          logger.log(line)
-        override def logContext: UIO[LogContext] =
-          logger.logContext
+  case class TestLogger(ref: Ref[Vector[(LogContext, String)]], logger: Logger) extends Logger {
```

Comment:
You could potentially turn this into a layer. Then you could provide a copy to each of your tests by just calling `provideCustomLayer` once on the whole suite. Then in your tests you could just call the methods you want with the environmental accessors and verify that you get the expected result without having to create the test logger within each tes.